interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -201,16 +240,28 @@ namespace cling {
       return;
     }
 
+
+    std::vector<std::unique_ptr<ASTConsumer>> Consumers;
+    HandlePlugins(*m_CI, Consumers);
+    std::unique_ptr<ASTConsumer> WrappedConsumer;
+
     DiagnosticsEngine& Diag = m_CI->getDiagnostics();
     if (m_CI->getFrontendOpts().ProgramAction != frontend::ParseSyntaxOnly) {
-      m_CodeGen.reset(CreateLLVMCodeGen(
+      m_CodeGen = CreateLLVMCodeGen(
```

Comment:
Could you capture the return of `CreateLLVMCodeGen()` in a `unique_ptr`, and then move that below, and set `m_CodeGen = CodeGen.get()`?

This signals ownership also within `IncrementalParser::IncrementalParser` and makes sure we handle ownership even if someone adds an early return in 2 years from now.